reframe/frontend/executors/policies.py

```diff
@@ -351,13 +389,15 @@ def runcase(self, case):
             if len(self._running_tasks[partname]) < partition.max_jobs:
                 # Task was put in _ready_tasks during setup
                 self._ready_tasks[partname].pop()
-                self._reschedule(task)
+                self._reschedule_compile(task)
+                self._reschedule_run(task)
             else:
                 self.printer.status('HOLD', task.check.info(), just='right')
-        except TaskExit:
+        except TaskExit: ## WHY?
```

Comment:
I can't find why we need this here so I am not sure if I am handling the exception correctly

Reply:
It is needed and it acts effectively as a "break" statement for a non-critical condition that that has happened in currently executing task. Check also #1331.